resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventProvider.java

```diff
@@ -146,10 +146,20 @@ public void write(int b) throws IOException
                      }
                      else
                      {
-                        if (escape && (b == '\n' || b == '\r' || b == '\\'))
+                        if (escape && b == '\n')
                         {
                            entityStream.write('\\');
-                           entityStream.write(b);
+                           entityStream.write('n');
```

Comment:
This used to write `\\` and `\n` while now it writes `\\` and `n` (similarly for `r`), is it really normal?

Reply:
Before the change, it actually doesn't escape the data. It still makes the data with "\n"  in two lines like : 
```
data: foo\
bar
```

Reply:
Hey Jim,

Look at SseEventInputImpl.readLine():
```
         if (!textLike && escape && b == '\\')
         {
            b = in.read();
            if (b != '\\' && b != '\n' && b != '\r')
            {
               throw new RuntimeException(Messages.MESSAGES.expectedExcapedCharacter(b));
            }
            out.write(b);
            continue;
         }
```
If it sees a '\\' followed by a line delimiter, it skips over the '\'. At the end of the data, there is an unescaped line delimiter.

-Ron

Reply:
@ronsigal  I changed SseEventProvider to escape the line delimiter: 
```
                        if (escape && b == '\n')
                        {
                           entityStream.write('\\');
                           entityStream.write('n');
                        }
                        else if (escape && b == '\r')
                        {
                           entityStream.write('\\');
                           entityStream.write('r');
                        }
                        else if (escape && b == '\\')
                        {
                           entityStream.write('\\');
                           entityStream.write('\\');
                        }
                        else
                        {
                           entityStream.write(b);
                        }
```
All the line delimiters will be escaped to ASCII characters. this will help to escape  "\" followed by a line delimiter ?